salt/states/file.py

```diff
@@ -7773,7 +7773,7 @@ def serialize(
                     "result": False,
                 }
 
-            with salt.utils.files.fopen(name, "r") as fhr:
+            with salt.utils.files.fopen(name, "rb") as fhr:
```

Comment:
trying to read a binary plist without the "b" would cause an exception. Also appears all the other implementations of  `salt.utils.files.fopen` in this file use `"rb"` as arguments.

Reply:
I moved this back to its previous behavior and only add a "b" if the formatter is a plist. this change was causing the `integration.states.test_file.FileTest.test_serializer_deserializer_opts` test to fail.